recommend/policyTemplates.go

```diff
@@ -0,0 +1,228 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Authors of KubeArmor
+
+package recommend
+
+import (
+	"archive/zip"
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/accuknox/auto-policy-discovery/src/types"
+	"github.com/cavaliergopher/grab/v3"
+	"github.com/google/go-github/github"
+	log "github.com/sirupsen/logrus"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	org   = "kubearmor"
+	repo  = "policy-templates"
+	url   = "https://github.com/kubearmor/policy-templates/archive/refs/tags/"
+	cache = ".cache/karmor/"
+)
+
+var currentVersion string
+
+func getCachePath() string {
+	return cache
+
+}
+
+// userHome function returns users home directory
+func userHome() string {
+	if runtime.GOOS == "windows" {
+		home := os.Getenv("HOMEDRIVE") + os.Getenv("HOMEPATH")
+		if home == "" {
+			home = os.Getenv("USERPROFILE")
+		}
+		return home
+	}
+	return os.Getenv("HOME")
+}
+
+// isTemplates function returns error if policy-template folder not found
+func isTemplates() error {
+	path := fmt.Sprintf("%s/%s", userHome(), getCachePath())
+	if stat, err := os.Stat(path); err == nil && stat.IsDir() {
+		if !isLatest() {
+			log.Warn("\033[1;33mA new version of policy-templates is available. Use `karmor recommend --update` to get recommendations based on the latest policy-templates.\033[0m")
+		}
+	} else {
+		log.Warn("\033[1;33mNo version of policy-templates available locally. Use `karmor recommend --update` to get recommendations based on the latest policy-templates.\033[0m")
+	}
```

Comment:
For both the cases, i.e if the local cache doesn't exists OR cache is outdated, throw the same message..
```
A new version of policy-templates is available. Use `karmor recommend update` to get recommendations based on the latest policy-templates.
```

Reply:
If we keep `version` as part of `rules.yaml`, then even while using embedded `rules.yaml`, we will be able to compare the current version with the latest one and decide if this warning needs to be shown.